catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/util/CrossPlatformFilePathEvaluator.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.ddf.commands.util;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import ddf.security.common.audit.SecurityLogger;
+
+/*
+ * Handles converting given Windows file paths (whose backslashes get escaped) into something the
+ * commands can handle.
+ */
+public class CrossPlatformFilePathEvaluator {
+
+    public static File handlePath(String path) {
```

Comment:
❓ Any reason we couldn't have used `Paths.get()`, `Path.isAbsolute()`, etc., instead of writing our own class? My assumption was that most of the OS specific differences in paths were taken care of by those new classes (see https://docs.oracle.com/javase/tutorial/essential/io/pathOps.html).

Reply:
The command gets the path string with the backslashes removed, so there's no way to tell the actual path structure without walking through the file system and trying to figure it out. For example, `C:\Users\greatUser\Downloads\example.txt` is received as `C:UsersgreatUserDownloadsexample.txt` and nothing can really figure out that's a path

Reply:
Are you saying that the `filePath` member variable in the command class (and therefore the `path` parameter here) doesn't have the `\`s in it when Karaf executes the command?

Reply:
Yes, they are removed before we get to any code in here.

Reply:
We should probably write a karaf bug for this. Can you get a valid path by escaping the `\` characters in the console?

Reply:
You can get a valid path by replacing `\` with `\\\\`. I'm not sure why it requires four backslashes to escape and get one...

Reply:
I have seen this before somewhere else, it is because it gets through a double interpretation. The first one replaces \\\\ with \\ and then the second one goes from \\ to \

Reply:
I made a Karaf issue https://issues.apache.org/jira/browse/KARAF-5320

Reply:
Thanks for the clarification @emanns95.